woocommerce/google-listings-and-ads
# API PULL - Block syncing when notifications are enabled

### Changes proposed in this Pull Request:

This PR disables Product and Coupon Sync when WC API is authorized. 

ℹ️  WC API is authorized  when the option `gla_wpcom_rest_api_status = approved`

### Detailed test instructions:

Enabled WC API 

0. Set the option `gla_wpcom_rest_api_status` as "approved"
1. Go to the Connection Test page
2. Verify you see  at the end WC STAtus APi is approved.
4. Run Sync All Products --- You will get "Error submitting products to Google: Manual product sync will not run if the automatic data fetching is enabled. Please review your configuration in Google Listing and Ads settings."
5. Run Delete All Synced Products --- You will get "Error deleting products from Google: Manual product sync will not run if the automatic data fetching is enabled. Please review your configuration in Google Listing and Ads settings."
6. Run Clean up All Synced Products --- You will get "Error cleaning up products: Manual product sync will not run if the automatic data fetching is enabled. Please review your configuration in Google Listing and Ads settings."
7. For the coupons there is no test in the Connection page available. So we can only test that the jobs (delete_coupon and update_coupon) are not scheduled when the coupon is created, updated or deleted. But this was tested already in previous works.

Disabled WC API 
1. Remove the option `gla_wpcom_rest_api_status` or click on "Disconnect" in the Connection test page under "WPCOM REST API Status" section
4. Run Sync All Products successfully 
5. Run Delete All Synced Products successfully 
6. Run Clean up All Synced Products successfully 
7. For the coupons there is no test in the Connection page available. So we can only test that the jobs (delete_coupon and update_coupon) are scheduled when the coupon is created, updated or deleted. But this was tested already in previous works.
